lib/web/join_tokens.go

```diff
@@ -245,6 +298,49 @@ func getJoinScript(settings scriptSettings, m nodeAPIGetter) (string, error) {
 	return buf.String(), nil
 }
 
+// generateIamTokenName makes a deterministic name for a iam join token
+// based on its rule set
+func generateIamTokenName(rules []*types.TokenRule) (string, error) {
+	// sort the rules by (account ID, arn)
+	// to make sure a set of rules will produce the same hash,
+	// no matter the order they are in the slice
+	orderedRules := make([]*types.TokenRule, len(rules))
+	copy(orderedRules, rules)
+	sortRules(orderedRules)
+
+	h := fnv.New32a()
+	for _, r := range orderedRules {
+		s := fmt.Sprintf("%s%s", r.AWSAccount, r.AWSARN)
+		_, err := h.Write([]byte(s))
+		if err != nil {
+			return "", err
+		}
+	}
+
+	return fmt.Sprintf("teleport-ui-iam-%d", h.Sum32()), nil
+}
+
+// sortRules sorts a slice of rules based on their AWS Account ID and ARN
+func sortRules(rules []*types.TokenRule) {
+	sort.Slice(rules, func(i, j int) bool {
+		accountID1, accountID2 := rules[i].AWSAccount, rules[j].AWSAccount
```

Comment:
nit: sometimes it can read better if you reuse the terms `i` and `j`.
```suggestion
		iAcct, jAcct := rules[i].AWSAccount, rules[j].AWSAccount
```